File: crates/auto-lang/test/a2r/017_struct_methods/struct_methods.expected.rs
struct Calculator {
    value: i32,
}

impl Calculator {
    fn add(&self, a: i32, b: i32) -> i32 {
        a + b
    }
    fn multiply(&self, a: i32, b: i32) -> i32 {
        a * b
    }
}

fn main() {
    let calc: Calculator = Calculator { value: 0 };
    let sum = calc.add(5, 3);
    let product = calc.multiply(4, 7);
    println!("Sum: {} {} {}", sum, "Product:", product);
}

File: crates/auto-lang/test/a2ts/017_struct_methods/struct_methods.expected.ts
class Calculator {
    value: number;

    constructor(value: number) {
        this.value = value;
    }

    add(a: number, b: number): number {
    a + b;
}

    multiply(a: number, b: number): number {
    a * b;
}
}

function main(): void {
    const calc = Calculator(0);
    const sum = calc.add(5, 3);
    const product = calc.multiply(4, 7);
    console.log("Sum:", sum, "Product:", product);
}

main();
